File: VB/DxBlazorApplication1/Services/Implementations/InMemoryLabelService.vb
Imports DxBlazorApplication1.Models
Imports DxBlazorApplication1.Services.DataHelpers
Imports DxBlazorApplication1.Services.Interfaces
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace ServiceLayer.Implementations.InMemory

    Public Class InMemoryLabelService
        Implements ILabelService(Of LectureType)

        Private ReadOnly lectureTypes As IList(Of LectureType)

        Public Sub New()
            lectureTypes = LabelDataHelper.GetLabels()
        End Sub

        Public Function GetLabelsAsync() As Task(Of IList(Of LectureType)) Implements ILabelService(Of LectureType).GetLabelsAsync
            Return Task.FromResult(lectureTypes)
        End Function
    End Class
End Namespace

File: CS/DxBlazorApplication1/Services/Implementations/InMemoryLabelService.cs
using DxBlazorApplication1.Models;
using DxBlazorApplication1.Services.DataHelpers;
using DxBlazorApplication1.Services.Interfaces;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceLayer.Implementations.InMemory {
    public class InMemoryLabelService : ILabelService<LectureType> {
        private readonly IList<LectureType> lectureTypes;

        public InMemoryLabelService() {
            lectureTypes = LabelDataHelper.GetLabels();
        }
        public Task<IList<LectureType>> GetLabelsAsync() {
            return Task.FromResult(lectureTypes);
        }
    }
}
